engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -165,7 +169,7 @@ public WorldRendererImpl(Context context, GLBufferPool bufferPool) {
         LocalPlayerSystem localPlayerSystem = context.get(LocalPlayerSystem.class);
         localPlayerSystem.setPlayerCamera(playerCamera);
 
-        renderableWorld = new RenderableWorldImpl(worldProvider, context.get(ChunkProvider.class), bufferPool, playerCamera, shadowMapCamera);
+        renderableWorld = new RenderableWorldImpl(worldProvider, context.get(ChunkProvider.class), bufferPool, playerCamera, dag);
```

Comment:
Hmmm... in light of what I suggested above I'd not pass the dag nor the shadowMapCamera. Just create a new setShadowMapCamera() method which get called on ShadowMapNode insertion into or removal from the graph. However, this would bring up the question what to do if there is no ShadowMapCamera set in the RenderableWorld instance? I guess the isChunkVisibleFromMainLight() method might want to throw an exception if no shadowmap camera has been set. But what consequences would that have elsewhere in the code? At this stage I don't know. Let me know what you think about this.

By the way, the RenderableWorldImpl instance effectively holds a number of acceleration structures, I'm tempted to call them caches but I'm not entirely sure it would be correct. One of these is related to shadow mapping, which is why we are having this conversation. EVENTUALLY, the ShadowMapNode (or any other node), probably on insertion in the graph, would want to have the chance to register the acceleration structures it needs. So, EVENTUALLY, the RenderableWorld class might either become a more generic acceleration structures manager (and be renamed accordingly) or, even better, acceleration structures become nodes in the pipeline.